### ISSUE TITLE
update to latest 7z version, currently 23.01

### ISSUE THREAD

**Issue by milahu** [p7zip-project/p7zip#225]
[7-zip.org](https://7-zip.org/download.html) says, the latest 7z version is 23.01

how come that the latest p7zip version is [16.02](https://sourceforge.net/projects/p7zip/files/p7zip/16.02/)?

**Comment by Cyberavater** [comment]
> result: the *.7z files have different checksums between 7z and 7zz

Afaik, the checksums **should** be different, because the header also contains time. You should be able to verify it by using the exact same tool and the same file.

**Comment by milahu** [comment]
> the header also contains time

no. when i change only the archive creation time (with `sleep 2`)
then the archive checksums are equal

when i `touch` a content file to change its file time (like `touch somedir/somefile`)
and then create a new archive, then the archive checksum changes

when everything else is equal, archives created with `7z` and `7zz`
differ in file size and about 0.5% of bytes

it would be interesting, why exactly these bytes are different...

<details>

```
$ du -b test-p7zip-17.05.1696022090.7z test-7zip-22.01.1696022170.7z 
25838   test-p7zip-17.05.1696022090.7z
25841   test-7zip-22.01.1696022170.7z
```

```
cat test-p7zip-17.05.1696022090.7z | xxd -p -c1 >test-p7zip-17.05.1696022090.7z.hex
cat test-7zip-22.01.1696022170.7z | xxd -p -c1 >test-7zip-22.01.1696022170.7z.hex
diff -u test-p7zip-17.05.1696022090.7z.hex test-7zip-22.01.1696022170.7z.hex
```


```diff
--- test-p7zip-17.05.1696022090.7z.hex
+++ test-7zip-22.01.1696022170.7z.hex
@@ -6,11 +6,11 @@
 1c
 00
 04
-f2
-b7
-7b
-0a
-aa
+1b
+70
+4a
+ca
+ad
 64
 00
 00
@@ -26,10 +26,10 @@
 00
 00
 00
-42
-36
-a3
-7d
+07
+3f
+79
+e1
 01
 63
 ff
@@ -25751,53 +25751,56 @@
 71
 4a
 b8
-66
-9b
+80
+39
+1e
 11
-c3
-2c
-cd
-fd
-a1
-dd
-54
-60
 bc
-d0
-33
-25
-33
-aa
-95
-9a
-50
-0a
-7d
-89
-e9
-78
+ba
+6d
+38
+c2
+d3
 a2
-f1
-b4
-b3
-d0
-fd
-1e
-c3
-c9
-de
-6f
-f3
+13
+41
+7e
+43
 25
+a3
+ea
+4a
+06
+53
+92
+00
+16
+59
 4b
-ff
-9a
-4e
-11
-c2
-08
+5f
+be
+80
+eb
+60
+64
+d0
+f7
+85
+79
+7e
+a1
+76
+1d
+df
+5a
+38
+69
+89
 7d
-f0
+75
+65
+54
+00
 00
 00
 17
@@ -25808,7 +25811,7 @@
 01
 09
 80
-a6
+a9
 00
 07
 0b
@@ -25830,9 +25833,9 @@
 66
 0a
 01
-ef
-d7
-f2
-4c
+bc
+97
+4b
+7e
 00
 00
```

</details>

surprising:

- `chown 0:0 somedir/somefile` has no effect with `7z` and `7zz`
- `chmod 0755 somedir/somefile` affects both `7z` and `7zz`

example:

```console
$ md5sum *.7z

bd6e72620782f82732e9baaccf35eded  test-7zip-22.01.1696022812.7z
bd6e72620782f82732e9baaccf35eded  test-7zip-22.01.1696022816.7z
bd6e72620782f82732e9baaccf35eded  test-7zip-22.01.1696022817.7z
bd6e72620782f82732e9baaccf35eded  test-7zip-22.01.1696022867.chown.7z
963cb9165a45764fe3292e9e8c55804c  test-7zip-22.01.1696022979.chmod.7z

bd6e72620782f82732e9baaccf35eded  test-7zip-23.01-7za.1696022824.7z
bd6e72620782f82732e9baaccf35eded  test-7zip-23.01-7za.1696022827.7z
bd6e72620782f82732e9baaccf35eded  test-7zip-23.01-7za.1696022829.7z
bd6e72620782f82732e9baaccf35eded  test-7zip-23.01-7za.1696022871.chown.7z
963cb9165a45764fe3292e9e8c55804c  test-7zip-23.01-7za.1696022990.chmod.7z

b23217c861ce849ec5eae0b5c96879d1  test-p7zip-17.05.1696022834.7z
b23217c861ce849ec5eae0b5c96879d1  test-p7zip-17.05.1696022837.7z
b23217c861ce849ec5eae0b5c96879d1  test-p7zip-17.05.1696022839.7z
b23217c861ce849ec5eae0b5c96879d1  test-p7zip-17.05.1696022874.chown.7z
94276534086d578c9fd51d67baaa4693  test-p7zip-17.05.1696022996.chmod.7z
```

**Comment by tansy** [comment]
And what's the problem?
Add long add archives produced are correct them difference between versions doesn't count.

**Comment by milahu** [comment]
> difference between versions doesn't count

yes, a difference does matter

> it would be interesting, why exactly these bytes are different

the question is: can we replace `p7zip/bin/7z` with `7z/bin/7zz`?
in other words: are `p7zip` and `7z` fully compatible?
in other words: is `7z/bin/7zz` a drop-in replacement for `p7zip/bin/7z`?

if the answer is yes, then linux distros can completely drop p7zip
and replace it with 7z, creating a symlink from `7z/bin/7zz` to `7z/bin/7z`

some arguments to keep p7zip were
"p7zip preserves ownership of files" (chown) or
"p7zip preserves permissions of files" (chmod) or
but `7z/bin/7zz` seems to store ownership and permissions of files

so, are there other reasons to keep p7zip?

**Comment by tansy** [comment]
There is branch 
[p7zip22.00](https://github.com/p7zip-project/p7zip/tree/p7zip22.00). You can make new, *p7zip23.01* branch and they may merge it into project as https://github.com/p7zip-project/p7zip/tree/p7zip23.01.

Before you request a pull change source new lines from Dos to Unix. You can use [this script](https://gist.github.com/tansy/c4a1f7ea1019915b4263c024757d3ac4).  Remember to follower the instruction - first copy while directory into backup (like `$ cp -r 7zip-23.01 7zip-23.01~`) and then `$ ./d2u-dir-1.sh <dir~> <dir>`.

**Comment by eclipseo** [comment]
> - Linux/macOS versions of 7-Zip: IUnknown interface in new code doesn't use 
>   virtual destructor that was used in previous 7-Zip and p7zip:
>      // virtual ~IUnknown() {}
>   So 7-Zip's dynamically linked shared libraries (codecs) are not compatible 
>   between new 7-Zip for Linux/macOS and old 7-Zip (and p7zip).

could that cause issues?

**Comment by tansy** [comment]
> could that cause issues?

Yes, incompatibility of pre-v23 standalone plugins in Linux/MacOS.
It had nothing to do with current p7zip here, that is v17, and very likely uses linked in plugins, so they don't have this issue.

In general that move is somewhat strange as standalone plugins from versions prior to v23 don't work anymore.
Sure to the fact that almost no-one uses then this easy no-one really cares, but, in general it's an issue.

**Comment by stefano2734** [comment]
Actual 7zip Version is 24.08

**Comment by cielavenir** [comment]
https://github.com/cielavenir/p7zip now ships 24.09

**Comment by cielavenir** [comment]
> You can make new, p7zip23.01 branch and they may merge it into project as https://github.com/p7zip-project/p7zip/tree/p7zip23.01.

This is what the maintainer is ought to do, asking for someone does not look right
